src/makelogline.c

```diff
@@ -144,7 +144,7 @@ void prepare_fixed_part(void) {
 	    ((strcmp(whichcontest, "qso") == 0) ||
 	     (strcmp(whichcontest, "dxped") == 0))) {
 	char khz[5];
-	sprintf(khz, " %3d", ((unsigned int)(freq / 1000.0)) % 1000);	// show freq.
+	sprintf(khz, " %3u", ((unsigned int)(freq / 1000.0)) % 1000);	// show freq.
```

Comment:
I'd prefer "%3d" with casting to int.
Unsigned int makes sense only if we'd like to log a 2+ GHz QSO on a 32 bit system. Quite unlikely.

A quick test for a potential negative double vs. unsigned int issue:
```
$ cat a.c
#include <stdio.h>

int main() {
    printf("%3d\n", (int)(12345.0/1000));
    printf("%3d\n", (int)(-12345.0/1000));
    printf("%3u\n", (int)(-12345.0/1000));
}

$ ./a.out 
 12
-12
4294967284

```

Reply:
Normally it would not make sense to have data type int for things which cannot be negative. Not having to test the lower border simplifies some tests.

But as we convert here from float anyway it would be good to keep the sign information.

Reply:
Funny you mention 2 GHz.  I having been considering a wishlist issue to bring Tlf out of the HF only world and be useful for VHF/UHF bands.  With the release of the Raspberry Pi 400 I have seen some hams write about using it for things like Field Day operations which in the USA permit operations on all bands though any bands above 23 cm are probably seldom used.

I see Tom prefers to simply cast to `int` and keep the `d` format specifier so that is what I'll go with.

Reply:
RPi 400 seems to be 64 bit, so one can tune until 9223372036 GHz, somewhere in the X-ray band. :-)

Reply:
I made the change to:

```
147         sprintf(khz, " %3d", ((int)(freq / 1000.0)) % 1000);    // show freq.
```

and am greeted with this from gcc 10.2.1:

```
../../tlf/src/makelogline.c: In function ‘prepare_fixed_part’:
../../tlf/src/makelogline.c:147:20: warning: ‘sprintf’ may write a terminating nul past the end of the destination [-Wformat-overflow=]
  147 |  sprintf(khz, " %3d", ((int)(freq / 1000.0)) % 1000); // show freq.
      |                    ^
../../tlf/src/makelogline.c:147:2: note: ‘sprintf’ output between 5 and 6 bytes into a destination of size 5
  147 |  sprintf(khz, " %3d", ((int)(freq / 1000.0)) % 1000); // show freq.
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

While clang 11.0.1 says nothing.  Hmmm.

Now I prefer my original patch of simply changing the conversion specifier.